File: fixtures/scale/typescript/large/src/checks/CheckMaskbyte.ts
import { Maskbyte } from '../ciphers/Maskbyte';
import { MaskbyteSpec } from '../specs/MaskbyteSpec';
import { CorpusMask } from '../corpus/CorpusMask';

export class CheckMaskbyte {
  static run(): boolean {
    const spec = MaskbyteSpec.get();
    for (const text of CorpusMask.texts()) {
      const sealed = Maskbyte.encrypt(text, spec.key);
      const opened = Maskbyte.decrypt(sealed, spec.key);
      if (opened !== text) return false;
    }
    return spec.category === 'mask';
  }
}

File: fixtures/scale/lua/large/checks/check_maskbyte.lua
local maskbyte = require("ciphers.maskbyte")
local maskbyte_spec = require("specs.maskbyte_spec")
local corpus_mask = require("corpus.corpus_mask")

local function check_maskbyte()
  local spec = maskbyte_spec.get()
  for _, text in ipairs(corpus_mask.corpus_mask()) do
    local sealed = maskbyte.maskbyte_encrypt(text, spec.key)
    local opened = maskbyte.maskbyte_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "mask"
end

return { check_maskbyte = check_maskbyte }
